### DISCUSSION TITLE
Text on top of cylinder

### DISCUSSION THREAD

**Discussion by Paul8043** [gumyr/build123d#551] [Q&A]
I have already learned how to write a text on the lateral of a cylinder. Now I have a similiar problem. How to write a text on the top of a cylinder. Each letter should
follow the direction the circumference of r1 (marked red in the sketch below).

![text-on-top](https://github.com/gumyr/build123d/assets/127483918/d9045e5c-4f3a-4944-8b98-01fe9bc034be)

This time the face is planar. The path might be circle r1? How to puzzle this together?

**Comment by gumyr** [comment]
```py
from build123d import *
from ocp_vscode import *

r2 = 100
r1 = 80
fs = 40

base_cylinder = Cylinder(r2, 300)
top = base_cylinder.faces().sort_by(Axis.Z)[-1]
text_path = (top - offset(top, r1 - r2)).face().inner_wires()[0]

text = Text(
    "around & around & around &", font_size=fs, path=text_path, position_on_path=0.45
)
embossed_text = extrude(text, 5)

show_all()
```
![image](https://github.com/gumyr/build123d/assets/85638518/5064c600-b9b2-404b-887d-3526ef1960d9)

**Reply by Paul8043** [reply]
Works perfectly. Thanks a lot, also for your quick response.
